airflow/providers/microsoft/winrm/hooks/winrm.py

```diff
@@ -199,7 +199,7 @@ def get_conn(self):
 
         # If endpoint is not set, then build a standard wsman endpoint from host and port.
         if not self.endpoint:
-            self.endpoint = 'http://{0}:{1}/wsman'.format(self.remote_host, self.remote_port)
+            self.endpoint = '{0}://{1}:{2}/wsman'.format(self.service.lower(), self.remote_host, self.remote_port)
```

Comment:
```suggestion
            self.endpoint = '{0}://{1}:{2}/wsman'.format(self.service, self.remote_host, self.remote_port)
```
Schema section of URL are case-insensitive and implementation should accept uppercase letters as equivalent to lowercase in scheme names (e.g., allow "HTTP" as well as "http"). See RFC 3986 for details.

Reply:
I 100% understand what you mean, but I believe it's best to use lowercase. The old behavious (pre-commit) was `http` lowercase. My goal is to keep the code behaving like it did before by default, to avoid regression (uppercase `HTTP` while RFC-compliant *is* a change in behaviour, thus it could have unexpected side-effects).

Reply:
@turbaszek , what is the recommended policy of the change management of Airflow project in this respect to use value in accordance with RFC without undue transformation? Should we change the default value to use "http" (now "HTTP" in the constructor)? Then transformation should not be necessary then.

Reply:
Hey @ad-m -> we have no policy for that. But If you would like to propose some - I invite you to start discussion at the devlist. This is community-driven project and You seem to be knowledgeable in this area, I would be very happy if someone like you could propose and discuss with the rest of the community any proposals you might have.

Reply:
That would be super-valuable contribution!